### PR TITLE
User-can-create-an-entry -- Tim Woo-Sam

### PR DESCRIPTION
**Pull request by twoosam**
Animation13

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -14,13 +14,72 @@
...
+                alt="placeholder" />
+            </div>
+            <div class="column-full column-half">
+              <label for="title">Title</label>
+              <input type="text" id="title" name="title" required />
+              <label for="photoURL">Photo URL</label>
+              <input
+                class="photo-input"
+                type="url"
+                id="photoURL"
```
Comment (shawnkost) [inline index.html]: `id` values should be `kebab-cased` rather than `camelCased`. Make sure to update the `label` as well

File: js/data.js [hunk 2]
```diff
@@ -6,3 +6,14 @@ let data = {
   editing: null,
   nextEntryId: 1,
 };
+
+function serialize() {
+  const dataModelJSON = JSON.stringify(data.entries);
```
Comment (shawnkost) [inline js/data.js]: You should be serializing the whole `data` object, rather than just `data.entries`

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const $photoInput = document.querySelector('.photo-input');
+function updateValue() {
+  document.getElementById('img').setAttribute('src', $photoInput.value);
```
Comment (shawnkost) [inline js/main.js]: You should be querying for DOM elements outside of functions. When you include it inside a function, you're forcing us to traverse the DOM over and over every single time this function gets called. If you just do it once globally, we can do the search once, and always have a reference to it

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    title,
+    photoURL,
+    notes,
+  };
+  formObject.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(formObject);
+  document
+    .getElementById('img')
+    .setAttribute('src', 'images/placeholder-image-square.jpg');
```
Comment (shawnkost) [inline js/main.js]: Same note as above about querying inside functions ^

**Review by shawnkost — CHANGES_REQUESTED** [review]
Nicely done so far Tim! A few changes requested here. One more thing I also noticed from your video

1. The `border` or `outline` should be purple rather than black around the `input` & `textarea`

**Comment by twoosam** [comment]
image
image

File: js/main.js [hunk 5]
```diff
@@ -1,14 +1,16 @@
 const $photoInput = document.querySelector('.photo-input');
+const $img = document.getElementById('img');
 function updateValue() {
-  document.getElementById('img').setAttribute('src', $photoInput.value);
+  $img.setAttribute('src', $photoInput.value);
 }
 $photoInput.addEventListener('input', updateValue);
 
 const $submit = document.querySelector('#form');
+const $imgPlaceholder = document.getElementById('img');
```
Comment (shawnkost) [inline js/main.js]: You've already queried for this on line 2, so you shouldn't do it again. Just utilize your `$img` variable

**Review by shawnkost — CHANGES_REQUESTED** [review]
One more small tweak here Tim!

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**